src/fileseq/utils.py

```diff
@@ -2,20 +2,45 @@
 """
 utils - General tools of use to fileseq operations.
 """
-from __future__ import absolute_import
+from __future__ import absolute_import, division
 
 from builtins import bytes
 from builtins import next
 from builtins import range
 from builtins import object
 import future.utils as futils
 
+import decimal
 import os
 from itertools import chain, count, islice
 
 from fileseq import exceptions
 
 
+_quantize_exponent_cache = {}
+
+
+def quantize(number, decimal_places, rounding=decimal.ROUND_HALF_EVEN):
+    """
+    Round a decimal value to given number of decimal places
+
+    Args:
+        number (decimal.Decimal): Decimal number to round
+        decimal_places (int): Number of decimal places in return value
+        rounding (str): decimal.Decimal rounding mode. See rounding argument of
+            https://docs.python.org/2/library/decimal.html#decimal.Context
+
+    Returns:
+        decimal.Decimal:
+    """
+    try:
+        quantize_exponent = _quantize_exponent_cache[decimal_places]
+    except KeyError:
+        quantize_exponent = decimal.Decimal(1).scaleb(-decimal_places)
+        _quantize_exponent_cache[decimal_places] = quantize_exponent
```

Comment:
Are we banking on the thread-safety issue not being a problem because of the cpython implementation? Or should we just add proper locking around this shared cache?

Reply:
The cache will always contain the same value for a given key, so it doesn't really matter?

Reply:
Ya I get that in cpython the assignment to the dict is effectively thread safe because of the GIL. I just have an initial reaction when I see global shared data structures being read and written to without a mutex :-) 
Its a nitpick.